### PR TITLE
Feature/bump nalgebra version to 0.32.1

### PR DESCRIPTION
**Pull request by geo-ant**
*Short*: increase nalgebra dependency to the latest version
*Changes*: changed deprecated method calls to the new view-API of nalgebra. That's all.

Please let me know if there's something I need to change.

*EDIT*: new changes due to updated clippy lints

### CODE REVIEW

**Comment by geo-ant** [comment]
hi, I should have fixed the failures inside the CI run. Could you kindly kick off the pipeline again?

File: src/utils.rs [hunk 1]
```diff
@@ -383,7 +383,7 @@ pub(crate) fn float_repr<F: Float>(f: F) -> alloc::string::String {
                 if i == 0 { "" } else { ":" }
             );
         }
-        out += &format!(" ({:.10E})", f);
+        out += &format!(" ({f:.10E})");
```
Comment (xd009642) [inline src/utils.rs:386]: My only comment would be these formatting changes may change the MSRV (beyond that of nalgebra update if it increases it as well). I'm not the main person on this repo so I'll leave it to @vadixidav to decide

Comment (geo-ant) [reply src/utils.rs:386]: Oh yeah, good point. I just blindly obeyed clippy's bidding...

Comment (xd009642) [reply src/utils.rs:386]: It may be fine, I'd hold off on reverting until someone else pitches in - plus it might be a lower MSRV than nalgebra anyway

Comment (geo-ant) [reply src/utils.rs:386]: Will do :)

**Comment by geo-ant** [comment]
Hi, just checking in to see what I can do to get this MR merged :)

**Comment by vadixidav** [comment]
This is fine, and I am not too worried about MSRV. Thank you for the reminder. I did discuss MSRV changes with @xd009642 a few days ago and never ended up performing the merge. I will publish a new release right away. A version bump will be performed regardless as nalgebra is part of the external facing API.

**Comment by vadixidav** [comment]
Released https://crates.io/crates/levenberg-marquardt/0.13.0

**Comment by geo-ant** [comment]
great, thank you kindly!

**Pull request merged**